### PR TITLE
Anniversaries page, View members kudos

### PR DESCRIPTION
**Pull request by arviens**

### CODE REVIEW

File: config.go [hunk 1]
```diff
@@ -10,18 +10,20 @@ import (
...
-	Channel             string `yaml:"channel"`
-	Database            string `yaml:"database"`
-	SlackCommandToken   string `yaml:"slack_command_token"`
-	AssetPath           string `yaml:"asset_path"`
-	CalendarID          string `yaml:"calendar_id"`
-	FullCalendarID      string `yaml:"full_calendar_id"`
-	CalendarSecret      string `yaml:"calendar_secret"`
-	CalendarCredentials string `yaml:"calendar_credentials"`
-	TemplatesPath       string `yaml:"templates_path"`
+	SlackToken					  string `yaml:"slack_token"`
```
Comment (mindaugas-dt) [inline config.go]: text formatting

File: database.go [hunk 2]
```diff
@@ -187,6 +187,32 @@ func loadKudosReceivedList() []KudosStats {
 	return statsList
 }
 
+func loadKudosReceivedByUser(userId string) []Kudo {
+	stmt, err := db.Prepare(`SELECT kudos.id, kudos.userFrom, kudos.message, kudos.value, kudos.date
```
Comment (mindaugas-dt) [inline database.go]: tables aliases for consistency?

File: templates/layouts/anniversaries.tmpl [hunk 3]
```diff
@@ -0,0 +1,27 @@
+{{ define "head" }}
+<meta http-equiv="refresh" content="30;/">
+<script src="https://code.highcharts.com/highcharts.js"></script>
+<script src="https://code.highcharts.com/modules/exporting.js"></script>
+{{ end }}
+{{ define "content" }}
+<h3 class="text-center color-white">{{.CurrentMonth}} work anniversaries:</h3>
+<div>
+    <div class="notes">
+    {{range .Employees}}
```
Comment (mindaugas-dt) [inline templates/layouts/anniversaries.tmpl:8]: What if range is empty?

File: templates/layouts/anniversaries.tmpl [hunk 4]
```diff
@@ -0,0 +1,27 @@
+{{ define "head" }}
+<meta http-equiv="refresh" content="30;/">
+<script src="https://code.highcharts.com/highcharts.js"></script>
```
Comment (mindaugas-dt) [inline templates/layouts/anniversaries.tmpl]: Why do you need highcharts in this page?

File: statistics.go [hunk 5]
```diff
@@ -0,0 +1,60 @@
...
+)
+
+type EmployeeAnniversary struct {
+	Name string
+	Date string
+	WorksYears int
+	Color string
+}
+
+func getWorkAnniversaries() [12][]EmployeeAnniversary {
```
Comment (mindaugas-dt) [inline statistics.go]: Why do you iterate through all 12 month if you only want to get information for one month?

File: statistics.go [hunk 6]
```diff
@@ -0,0 +1,60 @@
...
+
+	employees := [12][]EmployeeAnniversary{}
+	for i, row := range ws1.Rows {
+		if i == 0 {
+			continue
+		}
+
+		for j := 0; j < 12; j++ {
+			if row[j] != nil {
+				employeeWorkYear, _ := time.Parse("2006-01-02", r.FindString(row[j].Content))
```
Comment (mindaugas-dt) [inline statistics.go]: All data parsing should be in a separate function

File: statistics.go [hunk 7]
```diff
@@ -0,0 +1,60 @@
...
+				worksYears := time.Now().Year() - employeeWorkYear.Year()
+				employee := EmployeeAnniversary{Color: randomColor(),Date: r.FindString(row[j].Content), Name: strings.TrimSpace(rName.ReplaceAllString(row[j].Content, "")), WorksYears: worksYears}
```
Comment (mindaugas-dt) [inline statistics.go]: `row[j].Content` is used more than once, I think it should be extracted to separate variable

Comment (mindaugas-dt) [reply statistics.go]: I'm pretty sure this exceeds 120 symbols per line

File: page_member.go [hunk 8]
```diff
@@ -0,0 +1,31 @@
...
+
+	if !ok {
+		fmt.Errorf("The template %s does not exist.", pageName)
+	}
+
+	member, _ := findMemberByID(memberId)
+
+	pageData := memberPageView{
+		Member: member,
+		Kudos: loadKudosReceivedByUser(memberId)}
```
Comment (mindaugas-dt) [inline page_member.go]: text formatting

File: asset/c.css [hunk 9]
```diff
@@ -248,6 +258,49 @@ body.easter .note
...
 
+.stats-list .stats-list-row {
+  display: block;
+}
+.stats-list .stats-list-row:nth-child(n+6) {
+  display: none;
+}
+
+.show-list {
+  overflow-y: scroll;
```
Comment (mindaugas-dt) [inline asset/c.css:269]: I wonder how it looks like with a scrollbar :smile:

File: templates/layouts/member.tmpl [hunk 10]
```diff
@@ -0,0 +1,27 @@
...
+        <p>
+            <a href="/member/{{$.Member.ID}}" class="note-receiver-link">@{{$.Member.Name}}</a>,
+            <br>
+            {{.Text}}
+        </p>
+        <a href="/member/{{.MemberFrom.ID}}">@{{.MemberFrom.Name}}</a>
+      </div>
+    {{else}}
+    <span class="color-white">No kudos found</span>
+        {{end}}
```
Comment (mindaugas-dt) [inline templates/layouts/member.tmpl]: text formatting

File: templates/layouts/member.tmpl [hunk 11]
```diff
@@ -0,0 +1,27 @@
...
+    </div>
+</div>
+<style>
+body { overflow: auto; }
```
Comment (mindaugas-dt) [inline templates/layouts/member.tmpl]: Maybe this styling should be in header? You have empty definition at the top of this file
```
{{ define "head" }}{{end}}
```

**Review by mindaugas-dt — CHANGES_REQUESTED** [review]

File: asset/c.css (hunk 9 above)
Comment (arviens) [reply asset/c.css:269]: This is enabled only on specific click

**Review by mindaugas-dt — APPROVED** [review]

**Pull request merged**